iym/admin.py

```diff
@@ -0,0 +1,348 @@
+        data_package_name = f"national-in-year-spending-{financial_year}"
+        data_package_title = f"National in-year spending {financial_year}"
+
+        original_csv_path = unzip_uploaded_file(obj_to_update)
+
+        update_status(obj_to_update, "cleaning data")
+
+        csv_filename = os.path.basename(original_csv_path)
+
+        composite_key = create_composite_key_using_csv_headers(original_csv_path)
+
+        csv_table = tidy_csv_table(original_csv_path, composite_key)
+
+        update_status(obj_to_update, "uploading data")
+
+        func_result = create_data_package(csv_filename, csv_table, userid, data_package_name, data_package_title,
+                                          obj_to_update)
+
+        data_package = func_result['data_package']
+        datastore_token = func_result['datastore_token']
+
+        data_package_upload_authorisation = upload_data_package(data_package, userid, data_package_name,
+                                                                datastore_token,
+                                                                obj_to_update)
+
+        ##===============================================
+        # Starting import of uploaded data_package
+        update_status(obj_to_update, "import queued")
+        update_import_report(obj_to_update, "Starting import of uploaded datapackage.")
+        data_package_url = data_package_upload_authorisation["upload_url"]
+        status = import_uploaded_package(data_package_url, datastore_token, obj_to_update)
+
+        ##===============================================
+
+        check_and_update_status(status, data_package_url, obj_to_update)
+
+        os.remove(original_csv_path)
+
+        obj_to_update.process_completed = True
+        obj_to_update.save()
+    except Exception as e:
+        update_import_report(obj_to_update, str(e))
+        update_status(obj_to_update, "fail")
+
+
+class IYMFileUploadAdmin(admin.ModelAdmin):
+    readonly_fields = (
+        "import_report",
+        "user",
+        "process_completed",
+        "status"
+    )
+    fieldsets = (
+        (
+            "",
+            {
+                "fields": (
+                    "user",
+                    "financial_year",
+                    "latest_quarter",
+                    "file",
+                    "import_report",
+                    "status",
+                    "process_completed",
+                )
+            },
+        ),
+    )
+    list_display = (
+        "created_at",
+        "user",
+        "financial_year",
+        "latest_quarter",
+        "status",
+        "process_completed",
+        "updated_at",
+    )
+
+    def save_model(self, request, obj, form, change):
+        if not obj.pk:
+            obj.user = request.user
+        super().save_model(request, obj, form, change)
+
+        # async_task(func=process_uploaded_file, obj_id=obj.id)
+        process_uploaded_file(obj.id)
```

Comment:
it looks like it's processing synchronously, and this is probably why?

Reply:
this was a test commit - it's async now